org.openjkdf2.OpenJKDF2

shinyquagsire23 opened the pull request:
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission
____

Notes for Flathub Approval:
- I'm the project maintainer, I've been doing regular Win64/macOS releases [here](https://github.com/shinyquagsire23/OpenJKDF2/releases) for a while and Flatpak looks like a reasonable distribution method for Linux.
- [openjkdf2.org](https://openjkdf2.org/) is also mine.
- OpenJKDF2 is a game engine reimplementation similar to OpenRCT2, so it requires original game assets. No copyrighted assets are included in the distributable.

Permissions rundown:
- It's an SDL2 app, so OpenGL/HID controllers seem to need `--device=all`.
- On first startup, the app will prompt the user to locate a mounted install disk, or an existing game installation from GOG/Steam.
- If a copy of the game is found, assets are copied to `$XDG_DATA_HOME/openjkdf2/`.
- Player profiles, settings and savegames are stored in `$XDG_DATA_HOME/openjkdf2/player/`.
- Normally, OpenJKDF2 can run in a portable mode if it detects assets in the working directory (ie, on a flash drive or shared with another OS). For now, this will require a manual permission override.
- Mods are frequently installed by users, so it is expected that `$XDG_DATA_HOME/openjkdf2/` will be modified externally by those users.

Notes for Flatpak Team:
- Really really appreciate that cross-compiling is seamless and works on ARM64.
- Kinda wish there was a better-documented way to not run the x86_64 compiler through QEMU and use a cross-compiler instead. But eh, I get why that might not be the most practical.

hfiguiere commented on org.openjkdf2.OpenJKDF2.yml:
```diff
@@ -0,0 +1,32 @@
+app-id: org.openjkdf2.OpenJKDF2
+runtime: org.freedesktop.Platform
+runtime-version: '21.08'
```
```suggestion
runtime-version: '22.08'
```

hfiguiere commented on org.openjkdf2.OpenJKDF2.yml:
```diff
@@ -0,0 +1,32 @@
+command: openjkdf2-64
+
+modules:
+  - packaging/flatpak/glew/glew.json
+  - packaging/flatpak/glu/glu-9.json
```
please use the shared-modules for these two: https://github.com/flathub/shared-modules/

shinyquagsire23 replied:
Ah, I'd run into a certificate error with the glu-9 download, but it seems to be resolved now? Or it's cached somewhere and still broken I guess.

hfiguiere replied:
works fine everywhere else.

hfiguiere commented on org.openjkdf2.OpenJKDF2.yml:
```diff
@@ -0,0 +1,32 @@
+        url: https://github.com/shinyquagsire23/OpenJKDF2
+        commit: 421a3d927e756ad47820893aea0bfacb6e749772
+
+cleanup-commands:
```
modules is usually the last section

hfiguiere commented on org.openjkdf2.OpenJKDF2.yml:
```diff
@@ -0,0 +1,32 @@
+  - --socket=fallback-x11
+  - --socket=pulseaudio
+  - --filesystem=~/.local/share/openjkdf2:rw # Assets, mods, and saves are located here.
+  - --filesystem=host:ro # One-time asset installation prompts user for a disk (ie /run/media mounts) or an existing install (ie Steam, GOG).
```
this is too broad.

shinyquagsire23 replied:
Appreciate the pushback on this, I moved to GtkFileChooserNative and the sandboxing had no issues.

hfiguiere commented on org.openjkdf2.OpenJKDF2.yml:
```diff
@@ -0,0 +1,32 @@
+  - --share=network # needed for networking (Multiplayer/Valve GameNetworkingSockets)
+  - --socket=fallback-x11
+  - --socket=pulseaudio
+  - --filesystem=~/.local/share/openjkdf2:rw # Assets, mods, and saves are located here.
```
why don't you use XDG location instead?

hfiguiere commented on org.openjkdf2.OpenJKDF2.yml:
```diff
@@ -0,0 +1,32 @@
+app-id: org.openjkdf2.OpenJKDF2
```
openjkdf2.org is a placeholder. Is it own by the developer of that app?

shinyquagsire23 replied:
I'm the only developer of the app and the domain is mine, yes.

hfiguiere commented on org.openjkdf2.OpenJKDF2.yml:
```diff
@@ -0,0 +1,32 @@
+  - name: openjkdf2-64
+    buildsystem: simple
+    build-commands:
+      - bash packaging/flatpak/build_linux64_flatpak.sh
```
by Linux 64, you mean any 64-bits architecture, right?

shinyquagsire23 replied:
Yes, I test primarily on AArch64, but I also support x86_64. I guess 32-bit is technically supported though (I just don't test on it often) so I'll rename this.

hfiguiere replied:
we only support these two 64-bits arch.

I'm surprised that for once "linux64" doesn't equate x86_64. :-)

hfiguiere commented on org.openjkdf2.OpenJKDF2.yml:
```diff
@@ -0,0 +1,32 @@
+        commit: 421a3d927e756ad47820893aea0bfacb6e749772
+
+cleanup-commands:
+  - 'pip3 uninstall -y cogapp' # We only use this for building
```
install you should just `cleanup: '*'`. **in the module**

hfiguiere commented:
bot, build org.openjkdf2.OpenJKDF2

hfiguiere commented:
```
builddir/files/share/appdata/org.openjkdf2.OpenJKDF2.appdata.xml: FAILED:
• tag-missing           : <release> required
Validation of files failed
```

shinyquagsire23 commented:
Updated based on feedback:

- GtkFileChooserNative and portals are used now, instead of `--filesystem=host:ro`.
- Use `--filesystem=xdg-data:rw` instead of `--filesystem=~/.local/share/openjkdf2:rw`.
- Cleanup moved to the cogapp module.
- Renamed `linux64` -> `linux` and `openjkdf2-64` -> `openjkdf2` to avoid confusion on i386 builds.
- SDK 22.08 instead of 21.08 (the getting started documentation still references 21.08 just FYI)

hfiguiere commented on org.openjkdf2.OpenJKDF2.yml:
```diff
@@ -0,0 +1,28 @@
+  - --share=network # needed for networking (Multiplayer/Valve GameNetworkingSockets)
+  - --socket=fallback-x11
+  - --socket=pulseaudio
+  - --filesystem=xdg-data:rw # Assets, mods, and saves are located here.
```
you don't need this. It's automatically provided (and persisted)

shinyquagsire23 replied:
Fixed

barthalion commented:
bot, build org.openjkdf2.OpenJKDF2

barthalion commented:
/merge
